typing/includemod.ml

```diff
@@ -406,6 +406,19 @@ module Sign_diff = struct
     }
 end
 
+type 'a core_incl0 = 'a -> 'a -> (module_coercion, Error.sigitem_symptom) result
+type 'a core_incl =
+  loc:Location.t -> Env.t -> mark:mark -> Subst.t -> Ident.t -> 'a core_incl0
+type core_inclusion = {
+  value_descriptions: Types.value_description core_incl;
+  type_declarations: Types.type_declaration core_incl;
+  extension_constructors: Types.extension_constructor core_incl;
+  class_declarations: Env.t -> Subst.t -> Types.class_declaration core_incl0;
+  class_type_declarations:
+     loc:Location.t -> Env.t -> Subst.t ->
+     Types.class_type_declaration core_incl0;
```

Comment:
It is slightly weird that these methods have different signatures -- some of them need a location, but `class_declarations` does not, and the class ones do not use the `mark` parameter. Another choice would be to use the same `core_incl` type everywhere and then ignore the extra parameters when instantiating the record with the actual functions.

Reply:
I agree that having an uniform type is nice; I have decided to adapt the definition of the functions rather than have a closure adapter.